### ISSUE TITLE
DatePicker restores previous value on clicking outside of it's popper

### ISSUE THREAD

**Issue by dropik** [mui/mui-x#4567]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

If two DatePickers share the same value (for example stored as a state value), assigning a new value from the first DatePicker firstly seems to update the value of the second DatePicker. But actually, opening the popper of the second DatePicker and closing it, by clicking outside of it, makes the shared value to be restored to the previous value that was assigned once from the second DatePicker. Apparently, the problem occurs any time, when the value passed to the DatePicker gets updated somehow outside of that same DatePicker, sinche always in such circumstances opening the popper of that DatePicker and closing it restores value to the previous one.

### Expected behavior 🤔

I'd expect that rising shared state to the parent component would be more transparent, and that DatePickers would not memorize somehow internally previous values. Indeed, if two DatePickers would share the same value, updating it from the first DatePicker should update completely the second DatePicker as well, so that simply opening and closing the popper of the second DatePicker would not change that value.

### Steps to reproduce 🕹

Steps:

1. Implement a component with a Date state value (let it be today by default for example) and two simple DatePickers, that share that value. Launch it.
2. Set some new date from the first DatePicker.
3. Open the second DatePicker and close it by clicking outside.
4. The shared value is restored to the previous one.

### Context 🔦

I've encountered this issue in my project with one of DatePickers which was accepting a date value, stored in a Redux store. And this value can be updated as well from another place in my app. So it gets particularly annoing and unpredictable when opening and closing that DatePicker without changing the value resotres the value back to the previous one and changes lots of things in the app. I thought it was something in my code, by I looked to the DatePicker docs page and found the exact same issue on that page as well with 4 different DatePickers sharing same value.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  System:
    OS: Windows 10 10.0.19044
    CPU: (12) x64 AMD Ryzen 5 3600 6-Core Processor
    Memory: 8.31 GB / 15.92 GB
  Binaries:
    Node: 16.14.2 - C:\Program Files\nodejs\node.EXE
    npm: 8.7.0 - C:\Program Files\nodejs\npm.CMD
  Utilities:
    Git: 2.34.1.
  IDEs:
    VSCode: 1.66.2 - C:\Users\Painkiller\AppData\Local\Programs\Microsoft VS Code\bin\code.CMD
    Visual Studio: 17.0.32112.339 (Visual Studio Community 2022)
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (100.0.1185.39)
    Internet Explorer: 11.0.19041.1566
```
</details>

**Comment by JonathanGuo** [comment]
Having the same behaviour, I've also dug into the component `DesktopDateRangePicker `. It turns out the `usePickerState` hook has an `onDismiss` callback, which restores the internal state `initialDate` every time.

The only way to set the state is via the internal function `acceptDate`. Unfortunately, there is no external way to trigger the callback. No good workaround found yet except resetting the component key to force it re-render. (which shouldn't be the way)

**Comment by alexfauquette** [comment]
This issue should be solved by #4408

@dropik Can you confirm that on [this preview of the doc](https://deploy-preview-4408--material-ui-x.netlify.app/x/react-date-pickers/date-picker/#responsiveness) you can not reproduce the bug?

**Comment by dropik** [comment]
> This issue should be solved by #4408
> 
> @dropik Can you confirm that on [this preview of the doc](https://deploy-preview-4408--material-ui-x.netlify.app/x/react-date-pickers/date-picker/#responsiveness) you can not reproduce the bug?

Yes! On that doc page I can't reproduce the bug.
